Wood/Heat/Sand Land Upgrades and new weapons

akisephila opened the pull request:
Updates Wood Lands
Adds Silicone Caulk fluid, texture by Vinnelli
Various carved features
Carved Log and Carved Wooden Leaf, textured by Riotmode
Carved Bush by glubtier
Igneous Stone, Igneous Spike, and Pumice Stone from rose\_bushes_'s branch, textured by rose\_bushes_
Cindered Trees and woodset for Heat lands, trees from Rose's branch. Textures for planks and logs done collaboratively by Rose and Riotmode, textures for doors and trapdoors done by Akisephila
Updates textures for Chipboard and Wood Shavings. Textures done collaboratively by Vinnelli and Riotmode
Molten Amber fluid replacing lava in Heat lands, current texture by Akisephila temporary.
Sandy Grass, Dead Foliage, Tall Sandy Grass and Tall Dead Bush by rose\_bushes_, textured by rose\_bushes_
Treated and Lacquered variants of Carved Wood Blocks, textures done collaboratively by Vinnelli and Riotmode. Treated Planks retextured by Vinnelli. Lacquered Door and Trapdoor by Akisephila. Treated Ladder, Treated Bookshelf, Treated Door and Treated Trapdoor retextured by Akisephila to use Vinnelli's Treated Planks palette.
Singed Grass, Singed Foliage, and Sulfur Bubble from rose\_bushes_'s branch, textured by rose\_bushes_
Uncarved Wood Cruxite, Uranium, Emerald, Iron, and Redstone ores textured by Akisephila
Wooden Cactus retexture by Vinnelli
Wooden Grass, Treated Wooden Grass, Lacquered Wooden Mushroom, Wooden Lamp, Silicone Caulk Bucket and Molten Amber Bucket textured by Akisephila
Coffee Claws and TV Antenna from rose\_bushes_'s branch, textured by rose\_bushes_
Various Structures for Wood Lands built by Akisephila

kirderf1 commented on src/main/java/com/mraof/minestuck/block/plant/SpecialCactusBlock.java:
```diff
@@ -33,6 +34,6 @@ public boolean canSurvive(BlockState state, LevelReader level, BlockPos pos)
 	protected boolean isSustainableSoil(BlockState soil)
 	{
-		return soil.is(this) || soil.is(BlockTags.SAND);
+		return soil.is(this) || soil.is(MSTags.Blocks.WOOD_TERRAIN_BLOCKS) || soil.is(BlockTags.PLANKS);
```
Two things of note:
- The class name still signifies that the class can be used for other cacti blocks, but this change seems to be made specifically with the wooden cactus in mind. Something should be changed.
- The wooden cactus is still being used and placed on sand for the cactus pyramid center in sand land villages. Something should be changed.

akisephila replied:
wrt wooden cactus in sand lands, is that a specifically designed thing or does it call on whatever's under SpecialCactusBlock?

kirderf1 replied:
I'm not sure what you're asking.
A sand land structure is specifically placing some wooden cacti on the land surface block, which for such a land is sand.

akisephila replied:
SpecialCactusBlock is only used for Wooden Cactus. Change anyway or keep as is?

kirderf1 replied:
My quick suggestion for that is to just rename the class.

kirderf1 commented on CHANGELOG.md:
There are some things that I'm not seeing in the changelog. (I'm not sure if everything need to be added, but at least the more significant changes should)
- Cruxite block texture change
- Sand land features
- Oil has been renamed shale oil
- Coffee claws and tv antenna items

akisephila replied:
I wasn't aware the Cruxite Block texture was changed, think it was merged when I merged Rose's branch and I never noticed since there was no conflict.
Sand land features already listed in changelog, will update to be more clear

kirderf1 reviewed (changes requested):
It's a bit unfortunate that there's so much all in one PR, though I have no direct suggestion on how you might change your workflow. But I think a good starting point is to get a better understanding of git branches and how to use them to your benefit.

According to the CI, the `wood_terrain_blocks` block tag file need to be regenerated and committed.

With the large influx of new and related blocks, it'd be nice to have some separate classes for the registry objects and data generation (a la game object reorganization). But without any leading examples in the mod right now, I'm fine leaving it as is, and see if it can be done later.

With all the new and very similar template features being added when this PR gets merged, I'll have a good foundation for creating the configurable feature idea that I mentioned in #562.

akisephila commented:
Updated

kirderf1 commented:
Alright, great job on that!

Also, a reminder for myself to check tomorrow that no new textures or other assets are missing from the PR description credits.

akisephila commented on src/main/java/com/mraof/minestuck/block/MSHorizontalDirectionalBlock.java:
```diff
@@ -0,0 +1,48 @@
+/**
+ * A copy of {@link HorizontalDirectionalBlock} that is no longer abstract
+ */
+public class MSHorizontalDirectionalBlock extends HorizontalDirectionalBlock {
```
Yes

kirderf1 replied:
Alright. Then it might be a good idea to look over the models for the knotted wood blocks. Right now only the top and bottom textures change with the directional rotation while the sides remain the same, unlike the glazed terracotta.

akisephila replied:
yeah, noticed this but couldnt be bothered to try and fix it after the hassle it took to get it working as is

akisephila replied:
may just make them a directional like pillars or logs and have the top use the side texture.

akisephila commented on src/main/java/com/mraof/minestuck/block/MSBlockShapes.java:
```diff
@@ -31,6 +31,8 @@ public class MSBlockShapes
 	
 	public static final CustomVoxelShape SPIKES = new CustomVoxelShape(new double[]{0, 0, 0, 16, 4, 16}/*, new double[]{1, 1, 1, 15, 15, 15}*/);
 	
+	public static final CustomVoxelShape CARVED_WOODEN_LEAF = new CustomVoxelShape(new double[]{0, -1, 0, 16, 0, 16});
```
It was for hitbox reasons so that the player actually looks like they're standing on the block instead of above it

kirderf1 replied:
In that case, I suggest either:
- Raise just the min y value, putting it at 0.
- Raise not just the shape, but also the model.

kirderf1 replied:
Personally, I think that the second option seems nicer. (Which would visually be following the footsteps of vanilla blocks like ladders, rails, vines, and the glow lichen)

akisephila replied:
will do

akisephila commented:
> With these changes, should we perhaps move the wooden cactus from the main creative tab to the lands tab?
Can do
> Is the credit for planks, logs, doors and trapdoors in general, or specifically for cindered wood? Does textures for doors include both block textures and item textures?
Bit of a silly question? It's all in one sentence about the Cindered Woodset; why wouldn't it be specifically for Cindered Wood? And yes, textures for the blocks include item equivalents, and will include them unless stated otherwise
> Other assets that would be good to give credit in the PR description:
Most of these are included in commit, but will add to PR description anyway.

akisephila commented:
cruxite block retexture from glub's branch should be prioritized over temporary one here

kirderf1 commented:
All right. Removing the retexture from this PR seems appropriate then.

akisephila commented:
how, just delete the file in resources?

kirderf1 commented:
No, rather reverting it to the texture it was before this PR. (If you want, I could do it for you when this PR is otherwise ready to merge.)

akisephila commented:
> Alright, thanks for the added credits! Do you think you could add a clarification for the retextures of treated wood blocks? (that being bookshelves, ladders, doors and trapdoors) Also, could I hear your opinion on structure template credit?

done 
wrt structure template credit; i have no idea what that even is

kirderf1 commented:
By "structure templates", I'm referring to the `data/minestuck/structures/<...>.nbt` files. The credit would go to whoever designed those structures.

akisephila commented:
> By "structure templates", I'm referring to the `data/minestuck/structures/<...>.nbt` files. The credit would go to whoever designed those structures.

whoever invented tables and chairs, i guess? i built them all, but dont care much for credit since theyre mostly pretty basic

kirderf1 commented:
Alright. This is more about knowing where contributions come from than about recognition, so saying that you built all of them is good enough. (IMO recognition plays more into the [CREDITS.md](https://github.com/lunar-sway/minestuck/blob/1.20.1/CREDITS.md) file)

akisephila commented:
> No, rather reverting it to the texture it was before this PR. (If you want, I could do it for you when this PR is otherwise ready to merge.)

idk how to do that so that would be a help

akisephila commented:
also i started working on some heat land stuff before this feedback and itd be a hassle to delete it all before committing, is that fine? will add appropriate credit to description

kirderf1 commented:
> also i started working on some heat land stuff before this feedback and itd be a hassle to delete it all before committing, is that fine? will add appropriate credit to description

Okay so if you haven't committed it yet, I strongly recommend working with git so that you don't add it to this PR. At the simplest by "stashing/shelving" changes (a feature in git/intellij), and putting them back later. (If you're not sure what that is about, poke me on discord and we can work it out)

akisephila commented:
Updated

akisephila commented:
Updated

kirderf1 commented:
> Updated

I'm not seeing that second update.

akisephila commented:
wait neither am i what the fuck. i pushed the damn thing

akisephila commented:
just tried to push again. the fuck is happening? it just stopped working out of nowhere

Dweblenod commented on src/main/java/com/mraof/minestuck/block/SulfurBubbleBlock.java:
```diff
@@ -0,0 +1,21 @@
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+
+public class SulfurBubbleBlock extends TallGrassBlock
```
TallGrassBlock is not a suitable class to extend. BushBlock is not the best option either but ideally we will need something more custom (perhaps customized class can come after PR)

akisephila replied:
from rose's branch but understood

Dweblenod reviewed (commented):
There is an issue with the widest Features being cut off due to hard coded limitations of size related to chunk accessing. There is no straight forward solution so the advice of @kirderf1 is that the offending Features be removed for now. We can look for a solution later in order to re-add them!

Please fully remove these features:
- Sparse Large Unfinished Table
- Large Unfinished Table
- Massing Framing
- Massive Table

akisephila commented:
> There is an issue with the widest Features being cut off due to hard coded limitations of size related to chunk accessing. There is no straight forward solution so the advice of @kirderf1 is that the offending Features be removed for now. We can look for a solution later in order to re-add them!
> 
> Please fully remove these features:
> 
> * Sparse Large Unfinished Table
> * Large Unfinished Table
> * Massing Framing
> * Massive Table

I noticed this but feel as though it contributes to the personality of the land, making them unfinished. Most of the time they generate perfectly fine and not cut off anyway. In any case, removing these features would remove most of the personality of the land and the Furnish Jungle of rough biomes, which pretty much rely on these large structures. I say keep for now and if a solution is really necessary then it can be implemented later, but what's here now is fine especially as it causes no gameplay issues.

akisephila commented on src/main/java/com/mraof/minestuck/item/MSItems.java:
```diff
@@ -331,8 +335,8 @@ public class MSItems
 	public static final RegistryObject<Item> ACE_OF_DIAMONDS = REGISTER.register("ace_of_diamonds", () -> new WeaponItem(new WeaponItem.Builder(MSItemTypes.PAPER_TIER, 0, -1.8F).efficiency(0.0F).set(ItemRightClickEffect.switchTo(MSItems.CUESTICK)), new Item.Properties().defaultDurability(500)));
-	public static final RegistryObject<Item> ACE_OF_HEARTS = REGISTER.register("ace_of_hearts", () -> new Item(new Item.Properties().defaultDurability(500)));
-	
+	public static final RegistryObject<Item> TV_ANTENNA = REGISTER.register("tv_antenna", () -> new WeaponItem(new WeaponItem.Builder(Tiers.IRON, 3, -1.4F).efficiency(2.0F).set(MSItemTypes.MISC_TOOL).set(ItemRightClickEffect.switchTo(MSItems.ACE_OF_HEARTS)), new Item.Properties().defaultDurability(500)));
+	public static final RegistryObject<Item> ACE_OF_HEARTS = REGISTER.register("ace_of_hearts", () -> new WeaponItem(new WeaponItem.Builder(MSItemTypes.PAPER_TIER, 0, -1.8F).efficiency(0.0F).set(ItemRightClickEffect.switchTo(MSItems.TV_ANTENNA)), new Item.Properties().defaultDurability(500).rarity(Rarity.UNCOMMON)));
```
was part of rose's branch, will fix though

akisephila commented on src/main/java/com/mraof/minestuck/world/lands/terrain/WoodLandType.java:
```diff
@@ -24,63 +24,127 @@
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_TREE, LandBiomeType.NORMAL);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_TUNNEL, LandBiomeType.NORMAL);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.WOOD_SHAVINGS_PILE, LandBiomeType.NORMAL);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.WOODEN_CACTUS_PAIR, LandBiomeType.NORMAL);
 		
-		builder.addFeature(GenerationStep.Decoration.UNDERGROUND_ORES, MSPlacedFeatures.NETHERRACK_DISK,
+		builder.addFeature(GenerationStep.Decoration.UNDERGROUND_ORES, MSPlacedFeatures.TREATED_PLANKS_DISK,
 				FeatureModifier.withTargets(BlockPredicate.matchesBlocks(blocks.getBlockState("surface").getBlock(), blocks.getBlockState("upper").getBlock())), LandBiomeType.NORMAL);
+		builder.addFeature(GenerationStep.Decoration.UNDERGROUND_ORES, MSPlacedFeatures.CARVED_PLANKS_DISK,
+				FeatureModifier.withTargets(BlockPredicate.matchesBlocks(blocks.getBlockState("surface").getBlock(), blocks.getBlockState("upper").getBlock())), LandBiomeType.NORMAL);
+		builder.addFeature(GenerationStep.Decoration.UNDERGROUND_ORES, MSPlacedFeatures.CHIPBOARD_DISK,
+				FeatureModifier.withTargets(BlockPredicate.matchesBlocks(blocks.getBlockState("surface").getBlock(), blocks.getBlockState("upper").getBlock())), LandBiomeType.NORMAL);
+		
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.LACQUERED_WOODEN_MUSHROOM_PATCH, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.TREATED_WOODEN_GRASS_PATCH, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.CARVED_BUSH_PATCH, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.CARVED_CHERRY_TREE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.CARVED_LOG, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.SPARSE_LARGE_UNFINISHED_TABLE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.LARGE_UNFINISHED_TABLE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.SPARSE_TREATED_CHAIR, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.TREATED_CHAIR, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.SPARSE_TREATED_TABLE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.TREATED_TABLE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_BOOKSHELF, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.SPARSE_UNFINISHED_CARVED_CHAIR, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_CHAIR, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_CREEPER, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_DRAWER, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_SALAMANDER, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.SPARSE_UNFINISHED_CARVED_TABLE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_TABLE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_TREE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.UNFINISHED_CARVED_TUNNEL, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.WOOD_SHAVINGS_PILE, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.WOODEN_CACTUS_PAIR, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.MASSIVE_CHAIR, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.MASSIVE_FRAMING, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.MASSIVE_STOOL, LandBiomeType.ROUGH);
+		builder.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, MSPlacedFeatures.MASSIVE_TABLE, LandBiomeType.ROUGH);
```
understood

akisephila commented on src/main/java/com/mraof/minestuck/world/lands/terrain/WoodLandType.java:
```diff
@@ -24,63 +24,127 @@
 	@Override
 	public void registerBlocks(StructureBlockRegistry registry)
 	{
-		registry.setBlock("upper", Blocks.OAK_LOG);
```
understood

akisephila commented on src/main/java/com/mraof/minestuck/block/plant/BurnedFoliageBlock.java:
```diff
@@ -0,0 +1,24 @@
+	@Override
+	protected boolean mayPlaceOn(BlockState state, BlockGetter level, BlockPos pos)
+	{
+		return state.is(MSBlocks.IGNEOUS_STONE.get());
```
understood

akisephila commented on src/main/java/com/mraof/minestuck/block/plant/BurnedFoliageBlock.java:
```diff
@@ -0,0 +1,24 @@
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+
+public class BurnedFoliageBlock extends TallGrassBlock
```
will do

akisephila commented on src/main/java/com/mraof/minestuck/world/lands/terrain/HeatLandType.java:
```diff
@@ -39,9 +39,9 @@ public HeatLandType()
-		registry.setBlock("ground", Blocks.NETHERRACK);
-		registry.setBlock("upper", Blocks.NETHERRACK);
-		registry.setBlock("ocean", Blocks.LAVA);
+		registry.setBlock("ground", MSBlocks.PUMICE_STONE);
+		registry.setBlock("upper", MSBlocks.IGNEOUS_STONE);
```
already working on ores in shelved changes. will turn pumice into patches in heat land upgrades

akisephila commented on src/main/java/com/mraof/minestuck/data/recipe/MinestuckGristCostsProvider.java:
```diff
@@ -850,12 +852,33 @@ public static void buildRecipes(Consumer<FinishedRecipe> recipeSaver)
 		GristCostRecipeBuilder.of(MSBlocks.CAST_IRON.get()).grist(BUILD, 3).grist(RUST, 2).build(recipeSaver);
 		GristCostRecipeBuilder.of(MSBlocks.STEEL_BEAM.get()).grist(BUILD, 2).grist(RUST, 3).build(recipeSaver);
 		GristCostRecipeBuilder.of(MSBlocks.BLACK_SAND.get()).grist(BUILD, 2).grist(SULFUR, 1).build(recipeSaver);
+		GristCostRecipeBuilder.of(MSBlocks.IGNEOUS_STONE.get()).grist(TAR, 2).build(recipeSaver);
+		GristCostRecipeBuilder.of(MSBlocks.IGNEOUS_SPIKE.get()).grist(TAR, 1).build(recipeSaver);
+		GristCostRecipeBuilder.of(MSBlocks.PUMICE_STONE.get()).grist(TAR, 2).build(recipeSaver);
```
understood

akisephila commented on src/main/java/com/mraof/minestuck/data/worldgen/MSConfiguredFeatureProvider.java:
```diff
@@ -62,6 +62,28 @@ public static void register(BootstapContext<ConfiguredFeature<?, ?>> context)
 		context.register(PARCEL_PYXIS, new ConfiguredFeature<>(MSFeatures.PARCEL_PYXIS.get(), FeatureConfiguration.NONE));
 		context.register(FROG_RUINS, new ConfiguredFeature<>(MSFeatures.FROG_RUINS.get(), FeatureConfiguration.NONE));
 		
+		context.register(CARVED_CHERRY_TREE, new ConfiguredFeature<>(MSFeatures.CARVED_CHERRY_TREE.get(), FeatureConfiguration.NONE));
+		context.register(CARVED_HOUSE, new ConfiguredFeature<>(MSFeatures.CARVED_HOUSE.get(), FeatureConfiguration.NONE));
+		context.register(CARVED_LOG, new ConfiguredFeature<>(MSFeatures.CARVED_LOG.get(), FeatureConfiguration.NONE));
+		context.register(LARGE_CARVED_LOG, new ConfiguredFeature<>(MSFeatures.LARGE_CARVED_LOG.get(), FeatureConfiguration.NONE));
+		context.register(LARGE_UNFINISHED_TABLE, new ConfiguredFeature<>(MSFeatures.LARGE_UNFINISHED_TABLE.get(), FeatureConfiguration.NONE));
+		context.register(TREATED_CHAIR, new ConfiguredFeature<>(MSFeatures.TREATED_CHAIR.get(), FeatureConfiguration.NONE));
+		context.register(TREATED_TABLE, new ConfiguredFeature<>(MSFeatures.TREATED_TABLE.get(), FeatureConfiguration.NONE));
+		context.register(UNFINISHED_CARVED_BOOKSHELF, new ConfiguredFeature<>(MSFeatures.UNFINISHED_CARVED_BOOKSHELF.get(), FeatureConfiguration.NONE));
+		context.register(UNFINISHED_CARVED_CHAIR, new ConfiguredFeature<>(MSFeatures.UNFINISHED_CARVED_CHAIR.get(), FeatureConfiguration.NONE));
+		context.register(UNFINISHED_CARVED_CREEPER, new ConfiguredFeature<>(MSFeatures.UNFINISHED_CARVED_CREEPER.get(), FeatureConfiguration.NONE));
+		context.register(UNFINISHED_CARVED_DRAWER, new ConfiguredFeature<>(MSFeatures.UNFINISHED_CARVED_DRAWER.get(), FeatureConfiguration.NONE));
+		context.register(UNFINISHED_CARVED_SALAMANDER, new ConfiguredFeature<>(MSFeatures.UNFINISHED_CARVED_SALAMANDER.get(), FeatureConfiguration.NONE));
+		context.register(UNFINISHED_CARVED_TABLE, new ConfiguredFeature<>(MSFeatures.UNFINISHED_CARVED_TABLE.get(), FeatureConfiguration.NONE));
+		context.register(UNFINISHED_CARVED_TREE, new ConfiguredFeature<>(MSFeatures.UNFINISHED_CARVED_TREE.get(), FeatureConfiguration.NONE));
+		context.register(UNFINISHED_CARVED_TUNNEL, new ConfiguredFeature<>(MSFeatures.UNFINISHED_CARVED_TUNNEL.get(), FeatureConfiguration.NONE));
+		context.register(WOOD_SHAVINGS_PILE, new ConfiguredFeature<>(MSFeatures.WOOD_SHAVINGS_PILE.get(), FeatureConfiguration.NONE));
+		context.register(WOODEN_CACTUS_PAIR, new ConfiguredFeature<>(MSFeatures.WOODEN_CACTUS_PAIR.get(), FeatureConfiguration.NONE));
+		context.register(MASSIVE_CHAIR, new ConfiguredFeature<>(MSFeatures.MASSIVE_CHAIR.get(), FeatureConfiguration.NONE));
+		context.register(MASSIVE_FRAMING, new ConfiguredFeature<>(MSFeatures.MASSIVE_FRAMING.get(), FeatureConfiguration.NONE));
+		context.register(MASSIVE_STOOL, new ConfiguredFeature<>(MSFeatures.MASSIVE_STOOL.get(), FeatureConfiguration.NONE));
+		context.register(MASSIVE_TABLE, new ConfiguredFeature<>(MSFeatures.MASSIVE_TABLE.get(), FeatureConfiguration.NONE));
```
saw that was a possibility but couldnt find the time to figure out how. agree

akisephila commented:
Updated

kirderf1 commented:
> I noticed this but feel as though it contributes to the personality of the land, making them unfinished. Most of the time they generate perfectly fine and not cut off anyway. In any case, removing these features would remove most of the personality of the land and the Furnish Jungle of rough biomes, which pretty much rely on these large structures. I say keep for now and if a solution is really necessary then it can be implemented later, but what's here now is fine especially as it causes no gameplay issues.

The problem is about more than just structures being cut off. The size being too large also messes with the `TemplatePlacement` height functions, fairly often causing those structures to be placed in the ground at the bottom of the world.
But even if that wasn't the case or if we replaced the height check code, I still don't think those features should be accepted as-is.

To be clear, I am definitely fine with us taking inspiration from a bug and making it a feature. But I don't want us to use a bug as if it was a feature.

I still stand by my opinion that they should be removed from the PR for now, with the intention that someone rethinks and re-adds them after merging this (likely as a proper structure rather than a feature).

akisephila commented:
> > I noticed this but feel as though it contributes to the personality of the land, making them unfinished. Most of the time they generate perfectly fine and not cut off anyway. In any case, removing these features would remove most of the personality of the land and the Furnish Jungle of rough biomes, which pretty much rely on these large structures. I say keep for now and if a solution is really necessary then it can be implemented later, but what's here now is fine especially as it causes no gameplay issues.
> 
> The problem is about more than just structures being cut off. The size being too large also messes with the `TemplatePlacement` height functions, fairly often causing those structures to be placed in the ground at the bottom of the world.
> But even if that wasn't the case or if we replaced the height check code, I still don't think those features should be accepted as-is.
> 
> To be clear, I am definitely fine with us taking inspiration from a bug and making it a feature. But I don't want us to use a bug as if it was a feature.
> 
> I still stand by my opinion that they should be removed from the PR for now, with the intention that someone rethinks and re-adds them after merging this (likely as a proper structure rather than a feature).

i thought i already fixed them generating below the surface?

kirderf1 commented:
If you're talking about `VEGETAL_DECORATION` and `SURFACE_STRUCTURES`, those do not determine placement height. They're a sort of feature category which determine placement order.
If you're talking about something else, then I do not know what fix you're referring to.

akisephila commented:
Updated, not finished though

Dweblenod commented on src/main/java/com/mraof/minestuck/world/lands/terrain/WoodLandType.java:
```diff
@@ -20,67 +22,123 @@
-		registry.setBlock("light_block", MSBlocks.GLOWING_WOOD);
-		registry.setBlock("bush", Blocks.RED_MUSHROOM);
+		registry.setBlock("ground", MSBlocks.UNCARVED_WOOD);
+		registry.setBlock("upper", MSBlocks.CHIPBOARD);
```
I dont think it makes sense to have chipboard for upper since its sandwhiched between uncarved wood. Its okay to also make it uncarved

akisephila replied:
Fixed

Dweblenod reviewed (commented):
Wood  lands look pretty good! I would like the furniture to be more spread apart to reduce the frequency at which they overlap each other

akisephila commented:
Fixed Changelog conflicts. Currently fixing up structure spread a bit

akisephila commented:
Updated. Separation 4->5 and spacing 2->3. 4->6 on Separation was too much
